src/cmake/build_client.cmake

```diff
@@ -25,6 +25,10 @@ if(BUILD_CLIENT)
         if(OSXCROSS_HOST)
             message(STATUS "osxcross build, building main.m for client")
             list(APPEND mac_client_sources xcode/main.m)
+        else()
```

Comment:
Why _exactly_ is this in the `APPLE` block? Does CMake consider every BSD to be macOS?

Reply:
You might also be able to check for *BSD by comparing `CMAKE_HOST_SYSTEM_NAME` against the string `BSD`.

Reply:
What about Solaris/Illumos or Haiku, though

Reply:
My main concern is why it is supposed to work if it's in an `elseif(APPLE)` block.